File: js/euler46.js
function euler46(){
  var res = 0, n = 35;
  while (res == 0){
    let isSum = false;
    // n must be an odd composite
    if (isPrime(n)){
      n += 2;
      continue;
    }
    // n = p + 2r^2, n - 2r^2 = p
    for (let r = 1; 2*r*r < n; r++){
      // check if n - 2r^2 is prime
      if (isPrime(n - 2*r*r)){
        // n can be written as a sum of a prime and twice a square
        isSum = true;
        break;
      }
    }
    if (!isSum) res = n;
    n += 2;
  }
  return res;
}

function isPrime(n){
  if (n == 1) return false;
  for (let i = 2; i*i <= n; i++){
    if (n % i == 0) return false;
  }
  return true;
}

console.log(euler46());

File: go/euler46.go
package main
import "fmt"

func isPrime(n int) bool {
    if n < 2 {
        return false
    }
    if n == 2 {
        return true
    }
    for i := 2; i * i <= n; i++ {
        if n % i == 0 {
            return false
        }
    }
    return true
}

func main() {
    // 33 can be written as 2 * (1)^2 + 31
    res, n := 0, 35
    for res == 0 {
        // check subsequent odd composite numbers only
        if !isPrime(n) {
            foundSum := false
            for i := 1; i * i < n; i++ {
                s := i * i
                p := n - 2 * s
                // n can be written as the sum of a prime and twice a square
                if isPrime(p) {
                    foundSum = true
                    break
                }
            }
            // if no sum was found, n is a counterexample to the conjecture
            if !foundSum {
                res = n
            }
        }
        n += 2
    }
  fmt.Println(res)
}
